independent-projects/arc/runtime/src/main/java/io/quarkus/arc/CacheGet.java

```diff
@@ -0,0 +1,70 @@
+package io.quarkus.arc;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.inject.Instance;
+
+/**
+ * Instructs an injected {@link Instance} to cache the result of the {@link Instance#get()} operation
+ * <p>
+ * The result is "computed" on the first call to {@link Instance#get()} and the same value is returned for all subsequent calls,
```

Comment:
Maybe add a notion of being able to clear the cache?

Reply:
Good point.